### PR TITLE
Added Fp as sole auth method discussion and fix some typos

### PR DESCRIPTION
**Pull request by rickyyx**
Added : Line253 

https://github.com/cs3235-autoauth/report/compare/xc?expand=1#diff-cf15423ffa10955c5c3931e3981c8fa4R253

### CODE REVIEW

**Comment by Jemereny** [comment]
Looks good to me!

File: paper.tex [hunk 1]
```diff
@@ -208,27 +210,27 @@ \subsection{Fingerprinting for Authentication}
 
 \subsubsection{Desired Properties}
 In order to effectively augment user authentication with device fingerprinting, there are a number of properties that the device fingerprint should hold. A set of inappropriate fingerprinting practices may incur too many false negatives in the authentication process, denying legitimate users access or falling back to the most conservative authentication process too frequently. 
-Compared to the general usage of fingeprinting in advertising, fingerprinting for authentication requires a higher level of accuracy in distinguishing users. A $80\%$ or $90\%$ accuracy is simply not enough. \cite{alca:dev}
+Compared to the general usage of fingerprint in advertising, fingerprinting for authentication requires a higher level of accuracy in distinguishing users. A $80\%$ or $90\%$ accuracy is simply not enough. \cite{alca:dev}
```
Comment (Xenonym) [inline paper.tex:213]: fingerprints or fingerprinting

File: paper.tex [hunk 2]
```diff
@@ -238,16 +240,26 @@ \subsubsection{Fingerprint Authentication Solution}
...
+A simple replay attack can be effectively prevented by hashing the fingerprint at the client with a counter before sending to the server, as demonstrated by \cite{preu:smart}. However, for a more powerful attack such as XSS attack, it is less effective. The attacker can launch a XSS targeted at the victim, and steal the plain-text device fingerprint and the counter at the client-side. Once obtaining the fingerprint plain-text, the attacker can reproduce the hash by incrementing the counter
+and recomputing. In addition, such an attack is also possible even if the server dynamically changes the set of fingerprint attributes in each request \cite{unger:shpf} as the attacker can simply collect additional information about the target device once the hijacking is successful.
 
+\paragraph{ Fingerprint as the primary authentication method}
+The previous two approaches focus on how fingerprint can easily extend the current web authentication methods. We now present an approach where fingerprinting is used as the main authentication method, with supports from 2FA (like email, or SMS).
```
Comment (Xenonym) [inline paper.tex:254]: with _support_ (extra s)

like -> such as

File: paper.tex [hunk 3]
```diff
@@ -238,16 +240,26 @@ \subsubsection{Fingerprint Authentication Solution}
...
+A simple replay attack can be effectively prevented by hashing the fingerprint at the client with a counter before sending to the server, as demonstrated by \cite{preu:smart}. However, for a more powerful attack such as XSS attack, it is less effective. The attacker can launch a XSS targeted at the victim, and steal the plain-text device fingerprint and the counter at the client-side. Once obtaining the fingerprint plain-text, the attacker can reproduce the hash by incrementing the counter
+and recomputing. In addition, such an attack is also possible even if the server dynamically changes the set of fingerprint attributes in each request \cite{unger:shpf} as the attacker can simply collect additional information about the target device once the hijacking is successful.
 
+\paragraph{ Fingerprint as the primary authentication method}
+The previous two approaches focus on how fingerprint can easily extend the current web authentication methods. We now present an approach where fingerprinting is used as the main authentication method, with supports from 2FA (like email, or SMS). 
+Using fingerprint as the main authentication method requires a few assumptions to be held true:
```
Comment (Xenonym) [inline paper.tex:255]: fingerprints or fingerprinting

requires -> require (extra s)

File: paper.tex [hunk 4]
```diff
@@ -238,16 +240,26 @@ \subsubsection{Fingerprint Authentication Solution}
...
 
+\paragraph{ Fingerprint as the primary authentication method}
+The previous two approaches focus on how fingerprint can easily extend the current web authentication methods. We now present an approach where fingerprinting is used as the main authentication method, with supports from 2FA (like email, or SMS). 
+Using fingerprint as the main authentication method requires a few assumptions to be held true: 
+\begin{itemize}
+    \item Each device should have a unique fingerprint that can be associated with a user's account. From the previous sections results, while this is still not entirely true, it is believed that the potential huge number of attributes from a device fingerprint should set apart each user as more features can be collected from a device with more powerful fingerprinting methods. 
+    \item Fingerprints collected from the same device at different occasions need to be relatively stable, and there needs to be ways to re-associate a device to the account: recovery mechanism such as email or SMS can be viable options.
```
Comment (Xenonym) [inline paper.tex:258]: recovery mechanism_s_

File: paper.tex [hunk 5]
```diff
@@ -238,16 +240,26 @@ \subsubsection{Fingerprint Authentication Solution}
...
+    \item Fingerprints collected from the same device at different occasions need to be relatively stable, and there needs to be ways to re-associate a device to the account: recovery mechanism such as email or SMS can be viable options. 
+    \item Even though an attacker has full knowledge of the device's hardware and software information, the attacker should not be able to spoof the device if she/he does not have access to the device. 
+\end{itemize}
+
+If the above requirements can be met, device fingerprinting alone could be used in real life for account authentication. In the below sections, we will present a prototype that demonstrates the usage of fingerprint-based authentication method, and also discuss limitations that are closely related to the above requirements, and possible solutions to address them.
```
Comment (Xenonym) [inline paper.tex:262]: _the_ fingerprint-based authentication method

**Review by Xenonym — CHANGES_REQUESTED** [review]

**Pull request merged**